packages/lit-html/src/directives/class-map.ts

```diff
@@ -42,17 +43,30 @@ class ClassMapDirective extends Directive {
   }
 
   render(classInfo: ClassInfo) {
-    return Object.keys(classInfo)
-      .filter((key) => classInfo[key])
-      .join(' ');
+    // Add spaces to ensure separation from static classes
+    return (
+      ' ' +
+      Object.keys(classInfo)
+        .filter((key) => classInfo[key])
+        .join(' ') +
+      ' '
```

Comment:
The extra spaces are breaking the SSR tests (and the unit tests are failing first; I expect more to fail on the integration tests once the unit tests pass). Need to get those working.

FWIW, I don't feel strongly at all that we should support no-space rendering, but I assume this was supported in 1.x?

Reply:
separating the classes this way is the right semantic thing to do - we don't want the first class in the classMap to be smooshed together with a static style if the user forgot a space. I'll update the tests

Reply:
My reasoning was that the "idiom" with lit being strings is string concatenation (you can't do `class="foo bar${list.join(' ')}"` and get something good, so why should `class="foo bar${classMap({dynamic: true})}"`)...?

But it's fine, I see it's needed for consistency at least, since you'd render`bardynamic` initially, but try to remove only `dynamic` on subsequent updates, so it wouldn't be consistent.